server/controllers/comments.js

```diff
@@ -122,6 +122,204 @@ class CommentsController {
+                where: {
+                  id,
+                },
+                include: [{
+                  model: User,
+                  as: 'commenter',
+                  attributes: {
+                    exclude: ['hash', 'emailVerified', 'email', 'role', 'bio', 'twitter', 'linkedin', 'userFollowId', 'createdAt', 'updatedAt']
+                  }
+                }, {
+                  model: CommentLikesDislike,
+                  as: 'reactions',
+                  attributes: ['userId', 'username', 'commentId', 'reaction']
+                }]
+              })
+                .then(newLike => res.status(200).json(newLike))
+                .catch(next)).catch(next);
+          }).catch(next);
+      }).catch(next);
+  }
+
+  /**
+   * @description like or dislike comments made on articles
+   * @static
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @returns {next} next calls next
+   * @memberof CommentsController
+   */
+  static likeOrDislikeComment(req, res) {
+    if (req.params.reaction !== 'like' && req.params.reaction !== 'dislike') {
+      return res.status(400).json({
+        status: 'error',
+        message: 'You can only like or unlike this comment'
+      });
+    }
+    if (req.params.reaction === 'like') {
+      return CommentsController.commentReactionQuery(req, res, 'like', 'disliked', '0', '1');
+    }
+    if (req.params.reaction === 'dislike') {
+      return CommentsController.commentReactionQuery(req, res, 'dislike', 'liked', '1', '0');
+    }
+  }
+
+  /**
+   * @description get number of likes and dislikes in an article
+   * @static
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @param {object} next object
+   * @returns {next} res object
+   * @memberof CommentsController
+   */
+  static getCommentslikesandDislikes(req, res, next) {
+    const id = req.params.commentId;
+    const commentReactions = { likes: 0, dislikes: 0 };
+
+    Comment.findOne({
+      where: {
+        id,
+      }
+    })
+      .then((comment) => {
+        if (!comment) {
+          return res.status(404).json({
+            status: 'error',
+            message: 'comment does not exist',
+          });
+        }
+        CommentLikesDislike.findAll({
+          where: {
+            commentId: comment.id,
+          },
+        })
+          .then((commentLikes) => {
+            if (commentLikes === null) {
+              res.status(200).json({
+                status: 'success',
+                commentReactions: {
+                  likes: 0,
+                  dislikes: 0,
+                }
+              });
+            }
+            commentLikes.forEach((commentReaction) => {
+              if (commentReaction.reaction === '1') {
+                commentReactions.likes += 1;
+              } else if (commentReaction.reaction === '0') {
+                commentReactions.dislikes += 1;
```

Comment:
a better implementation is to find all where reaction === 0 or reaction === 1 and then total number of likes would be the length of the returned value.

Reply:
Thank you. I will also work on this